app/scripts/modules/amazon/src/loadBalancer/configure/common/LoadBalancerLocation.tsx

```diff
@@ -156,14 +156,10 @@ export class LoadBalancerLocation extends React.Component<ILoadBalancerLocationP
       ([allSubnets, subnetPurpose]) => allSubnets && allSubnets.find(subnet => subnet.purpose === subnetPurpose),
     );
 
-    // I don't understand why we use subnet.availabilityZones here, but region.availabilityZones below.
```

Comment:
@vigneshm do you understand why we used subnet.availabilityZones here, but region.availabilityZones below?

Reply:
I found two things related to these changes.

1. `availabilityZones` from `account.region` do not get used at all here. `AvailabilityZoneSelector.tsx` overrides the selected zones when mounted and only makes the availability zones (computed from subnets) up for selection.
2. @robzienert did confirm in a slack thread that this (i.e computing availability zones from subnets) is the expected behavior here.

Reply:
+1 -- I would say calculating AZs from subnets over regions is always the correct decision, especially if dealing with networking.